algod/Dockerfile

```diff
@@ -0,0 +1,72 @@
+ARG GO_VERSION=1.17.5
```

Comment:
It's an ARG, so you can change it when calling `docker build`.

It probably makes more sense for this to live in go-algorand so that the correct version can be set according to what branch is running.